### PR TITLE
Fix Xcode 14 and Swift 5.7 compiler error

### PR DESCRIPTION
**Pull request by dangthaison91**
https://github.com/danielgindi/Charts/pull/4823#issuecomment-1195361015

### Issue Link :link:
Fix compilation error with Xcode 14 and Swift 5.7
#4863 #4870 #4838 #4835

### Implementation Details :construction:
This PR simply collect other fixes by several developers into one PR:
- #4823 
- #4836
- https://github.com/danielgindi/Charts/issues/4860#issuecomment-1178968435

### CODE REVIEW

**Comment by dingtianran** [comment]
I can confirm this patch PR works for xcode 14 beta 3&4

File: Source/Charts/Animation/ChartAnimationEasing.swift [hunk 1]
```diff
@@ -341,7 +341,7 @@ internal struct EasingFunctions
         let s: TimeInterval = 1.70158
         var position: TimeInterval = elapsed / duration
         position -= 1.0
-        return Double( position * position * ((s + 1.0) * position + s) + 1.0 )
+        return Double( position * position * ((s + Double(1.0)) * position + s) + Double(1.0) )
```
Comment (iDevPro) [inline Source/Charts/Animation/ChartAnimationEasing.swift:344]: Would be better if `s` and `position` typecasted before calculate ?
Or 1.0 would be typecasted once ?)

File: Source/Charts/Renderers/CandleStickChartRenderer.swift [hunk 2]
```diff
@@ -38,9 +38,9 @@ open class CandleStickChartRenderer: LineScatterCandleRadarRenderer
             accessibleChartElements.append(element)
         }
 
-        for case let set as CandleChartDataSetProtocol in candleData where set.isVisible
+        for case let set as CandleChartDataSetProtocol in (candleData as ChartData) where set.isVisible
```
Comment (iDevPro) [inline Source/Charts/Renderers/CandleStickChartRenderer.swift:41]: set is not useful name for it, because set is special word in swift )

**Review by vmagnani — APPROVED** [review]

File: Source/Charts/Renderers/CandleStickChartRenderer.swift (hunk 2 above)
Comment (dangthaison91) [reply Source/Charts/Renderers/CandleStickChartRenderer.swift]: I changed to :
`for case let dataSet as CandleChartDataSetProtocol in (candleData as ChartData) where dataSet.isVisible`

File: Source/Charts/Animation/ChartAnimationEasing.swift (hunk 1 above)
Comment (dangthaison91) [reply Source/Charts/Animation/ChartAnimationEasing.swift]: Updated, just remove Double cast and Xcode can build the project @iDevPro

**Review by iDevPro — APPROVED** [review]

**Review by mariuspatru — APPROVED** [review]

**Comment by crazymanish** [comment]
A friendly ping to @pmairoldi to include this PR in the upcoming release! 🙏

**Comment by alanzeino** [comment]
In case anyone is wondering, beta 5 will likely be very similar to the 14.0 GM or public version, so this (or #4881) would be ready to land or use in your local fork of Charts.

**Comment by waterskier2007** [comment]
Has anyone tested this on lower version of the OS such as 14.5? I am getting some crashes when applying this fix to my own fork.

**Comment by iDevPro** [comment]
> Has anyone tested this on lower version of iOS such as 14.5? I am getting some crashes when applying this fix to my own fork.

Which crashes ? May you show gist ?

**Comment by adityarajpt** [comment]
> Has anyone tested this on lower version of iOS such as 14.5? I am getting some crashes when applying this fix to my own fork.

@waterskier2007 I tried on 14.5, I don't see any crash so far

**Comment by werkaflute** [comment]
On XCode 14.6 Beta I have errors :/

**Comment by pmairoldi** [comment]
Everything from this PR has been merged from other PRs. Thanks contributing!

**Comment by nitinkarki** [comment]
@pmairoldi When will there be a new release or tag with all the changes in `master`? Specifically the Xcode 14 support

**Comment by pmairoldi** [comment]
When Xcode 14 gm is released

**Comment by alanzeino** [comment]
> @pmairoldi @danielgindi When will there be a new release or tag with all the changes in `master`? Specifically the Xcode 14 support

change your package manager to point to the git commit and you don't need to wait for a new release or tag

**Comment by PorterHoskins** [comment]
@danielgindi Is there a plan to change the package name so we can use Apple's Charting library?

**Pull request closed without merging**